// @tanstack/vue-query-devtools/dist/cjs/devtools.vue.cjs
"use strict";
const vue = require("vue");
const vueQuery = require("@tanstack/vue-query");
const queryDevtools = require("@tanstack/query-devtools");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "devtools",
  props: {
    initialIsOpen: { type: Boolean },
    buttonPosition: {},
    position: {},
    client: {},
    errorTypes: {},
    styleNonce: {},
    shadowDOMTarget: {}
  },
  setup(__props) {
    const props = __props;
    const div = vue.ref();
    const client = props.client || vueQuery.useQueryClient();
    const devtools = new queryDevtools.TanstackQueryDevtools({
      client,
      queryFlavor: "Vue Query",
      version: "5",
      onlineManager: vueQuery.onlineManager,
      buttonPosition: props.buttonPosition,
      position: props.position,
      initialIsOpen: props.initialIsOpen,
      errorTypes: props.errorTypes,
      styleNonce: props.styleNonce,
      shadowDOMTarget: props.shadowDOMTarget
    });
    vue.watchEffect(() => {
      devtools.setButtonPosition(props.buttonPosition || "bottom-right");
      devtools.setPosition(props.position || "bottom");
      devtools.setInitialIsOpen(props.initialIsOpen);
      devtools.setErrorTypes(props.errorTypes || []);
    });
    vue.onMounted(() => {
      devtools.mount(div.value);
    });
    vue.onScopeDispose(() => {
      devtools.unmount();
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", {
        className: "tsqd-parent-container",
        ref_key: "div",
        ref: div
      }, null, 512);
    };
  }
});
module.exports = _sfc_main;
