// @tanstack/vue-query-devtools/dist/esm/devtools.vue.js
import { defineComponent, ref, watchEffect, onMounted, onScopeDispose, openBlock, createElementBlock } from "vue";
import { useQueryClient, onlineManager } from "@tanstack/vue-query";
import { TanstackQueryDevtools } from "@tanstack/query-devtools";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "devtools",
  props: {
    initialIsOpen: { type: Boolean },
    buttonPosition: {},
    position: {},
    client: {},
    errorTypes: {},
    styleNonce: {},
    shadowDOMTarget: {}
  },
  setup(__props) {
    const props = __props;
    const div = ref();
    const client = props.client || useQueryClient();
    const devtools = new TanstackQueryDevtools({
      client,
      queryFlavor: "Vue Query",
      version: "5",
      onlineManager,
      buttonPosition: props.buttonPosition,
      position: props.position,
      initialIsOpen: props.initialIsOpen,
      errorTypes: props.errorTypes,
      styleNonce: props.styleNonce,
      shadowDOMTarget: props.shadowDOMTarget
    });
    watchEffect(() => {
      devtools.setButtonPosition(props.buttonPosition || "bottom-right");
      devtools.setPosition(props.position || "bottom");
      devtools.setInitialIsOpen(props.initialIsOpen);
      devtools.setErrorTypes(props.errorTypes || []);
    });
    onMounted(() => {
      devtools.mount(div.value);
    });
    onScopeDispose(() => {
      devtools.unmount();
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        className: "tsqd-parent-container",
        ref_key: "div",
        ref: div
      }, null, 512);
    };
  }
});
export {
  _sfc_main as default
};
